backend/internal/nucleusdb/json-models/models.go

```diff
@@ -132,29 +132,158 @@ type JobMapping struct {
 	Schema      string
 	Table       string
 	Column      string
-	Transformer string
+	Transformer *Transformer
 	Exclude     bool
 }
 
 func (jm *JobMapping) ToDto() *mgmtv1alpha1.JobMapping {
+
 	return &mgmtv1alpha1.JobMapping{
 		Schema:      jm.Schema,
 		Table:       jm.Table,
 		Column:      jm.Column,
-		Transformer: jm.Transformer,
+		Transformer: jm.Transformer.ToDto(),
 		Exclude:     jm.Exclude,
 	}
 }
 
 func (jm *JobMapping) FromDto(dto *mgmtv1alpha1.JobMapping) error {
+
+	t := &Transformer{}
+	if err := t.FromDto(dto.Transformer); err != nil {
+		return err
+	}
 	jm.Schema = dto.Schema
 	jm.Table = dto.Table
 	jm.Column = dto.Column
-	jm.Transformer = dto.Transformer
+	jm.Transformer = t
 	jm.Exclude = dto.Exclude
 	return nil
 }
 
+type Transformer struct {
+	Value  string
+	Config *TransformerConfigs
+}
+
+type TransformerConfigs struct {
+	EmailConfig *EmailConfigs
+	FirstName   *FirstNameConfig
+	Uuidv4      *Uuidv4Config
+	PhoneNumber *PhoneNumberConfig
+	Passthrough *PassthroughConfig
+}
+
+type EmailConfigs struct {
+	PreserveLength bool
+	PreserveDomain bool
+}
+
+type FirstNameConfig struct {
+}
+type Uuidv4Config struct {
+}
+type PhoneNumberConfig struct {
+}
+type PassthroughConfig struct {
+}
+
+// from API -> DB
+func (t *Transformer) FromDto(tr *mgmtv1alpha1.Transformer) error {
+
+	switch tr.Config.Config.(type) {
+	case *mgmtv1alpha1.TransformerConfig_EmailConfig:
+		t.Value = tr.Value
+		t.Config = &TransformerConfigs{
+			EmailConfig: &EmailConfigs{
+				PreserveLength: tr.Config.GetEmailConfig().PreserveLength,
+				PreserveDomain: tr.Config.GetEmailConfig().PreserveDomain,
+			},
+		}
+	case *mgmtv1alpha1.TransformerConfig_FirstNameConfig:
+		t.Value = tr.Value
+		t.Config = &TransformerConfigs{
+			FirstName: &FirstNameConfig{},
+		}
+	case *mgmtv1alpha1.TransformerConfig_PassthroughConfig:
+		t.Value = tr.Value
+		t.Config = &TransformerConfigs{
+			Passthrough: &PassthroughConfig{},
+		}
+	case *mgmtv1alpha1.TransformerConfig_UuidConfig:
+		t.Value = tr.Value
+		t.Config = &TransformerConfigs{
+			Uuidv4: &Uuidv4Config{},
+		}
+	case *mgmtv1alpha1.TransformerConfig_PhoneNumberConfig:
+		t.Value = tr.Value
+		t.Config = &TransformerConfigs{
+			PhoneNumber: &PhoneNumberConfig{},
+		}
+	default:
+		return fmt.Errorf("invalid config")
```

Comment:
If a transformer doesn't have a config, I think the default should just be: 
```
t.Value = tr.Value
t.Config = &TransformerConfigs{}
```
Then not ever transformer needs a config.